pandas/core/reshape/pivot.py

```diff
@@ -627,11 +632,19 @@ def _normalize(table, normalize, margins, margins_name="All"):
             column_margin = column_margin / column_margin.sum()
             table = concat([table, column_margin], axis=1)
             table = table.fillna(0)
+            table.columns = table_columns
 
         elif normalize == "index":
-            index_margin = index_margin / index_margin.sum()
+            # index_margin is a dataframe, and use a hacky way: sum(axis=1)[0]
+            # to get the normalized result, and use sum() instead for series
+            if isinstance(index_margin, ABCDataFrame):
```

Comment:
why do you need to distinguish MI here?

Reply:
the point is, if MI, then after pivoting, `index_margin` is a `DataFrame`, while for Index, `index_margin` is a `Series`, and the way to calculate normalized value is to divide those by the sum, and for `Series` it's easy because the `sum` is number, but situation for `DF` is different, so i have to distinguish. right now, this method doesn't seem welcome to MI situation. @jreback

Reply:
yes, after i changed the code above, this issue can be solve as well. And i removed this if/then. thank you for pointing it out @jreback